client/components/form.jsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+
+export default class SignUpForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      username: '',
+      password: '',
+      isLoggedIn: false
+    };
+    this.handleUsernameChange = this.handleUsernameChange.bind(this);
+    this.handlePasswordChange = this.handlePasswordChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleUsernameChange(event) {
+    this.setState({
+      username: event.target.value
+    });
+  }
+
+  handlePasswordChange(event) {
+    this.setState({
+      password: event.target.value
+    });
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    const newUser = {
+      username: this.state.username,
+      password: this.state.password
+    };
+
+    fetch('/api/auth/sign-up', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(newUser)
+    })
+      .then(response => response.json())
+      .then(data => {
+        this.setState({
+          username: '',
+          password: '',
+          isLoggedIn: true
```

Comment:
We do not use a boolean for if the user is logged in, we use the token from local storage. also, you can remove this because the feature is only for signing up.